src/java/tring/src/main/java/io/privacyresearch/tring/TringServiceImpl.java

```diff
@@ -627,4 +638,48 @@ private void executeRequest(Runnable r) {
         Future<?> submit = executor.submit(r);
         LOG.info("Execution state = " + submit.state());
     }
+    
+    @Override
+    public byte[] getCallLinkBytes(String link) {
+        try {
```

Comment:
It seems that the shared arena (the variable `scope`) is never closed. Hence it looks to me that all allocations are never freed? Would it not make more sense to create a specific shared arena within the boundaries of this method?

```
try (Arena arena = Arena.ofShared()) {
    // do things with arena
} // the arena gets closed automatically at this point, which also releases any off-heap allocations
```